lib/charms/sdcore_nms_k8s/v0/fiveg_core_gnb.py

```diff
@@ -0,0 +1,402 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the `fiveg_core_gnb` relation.
+
+This library contains the Requires and Provides classes for handling the `fiveg_core_gnb`
+interface.
+
+The purpose of this library is to provide a way for a FiveG core to provide network information and
+configuration to CUs/gNodeBs.
+
+To get started using the library, you need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.sdcore_nms_k8s.v0.fiveg_core_gnb
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- pydantic
+- pytest-interface-tester
+
+Charms providing the `fiveg_core_gnb` relation should use `FivegCoreGnbProvides`.
+Typical usage of this class would look something like:
+
+    ```python
+    ...
+    from charms.sdcore_gnbsim_k8s.v0.fiveg_core_gnb import FivegCoreGnbProvides
+    ...
+
+    class SomeProviderCharm(CharmBase):
+
+        def __init__(self, *args):
+            ...
+            self.fiveg_core_gnb_provider = FivegCoreGnbProvides(
+                charm=self,
+                relation_name="fiveg_core_gnb"
+                )
+            ...
+            self.framework.observe(
+                self.fiveg_core_gnb_provider.on.fiveg_core_gnb_request,
```

Comment:
There's no such event in the `FivegCoreGnbProvides` class. In our case, it's gonna be the Pebble notice that we listen to.
To be independent from the implementation, we probably shouldn't be defining any specific event here.